lib/dictBuilder/cover.c

```diff
@@ -40,6 +40,11 @@
 /*-*************************************
 *  Constants
 ***************************************/
+/**
+* There are 32bit indexes used to ref samples, so limit samples size to 4GB
+* on 64bit builds.
+* For 32bit builds 1 GB is an arbitrary memory limit.
```

Comment:
That's a good thing to add some comments to clarify the intent of the code ! 👍 

A word of comment on the "1 GB limit for 32-bit" : 
it's worth knowing that we typically use a few "large" allocations, which require continuous memory segments.
In 32-bit address space, this quickly becomes a problem, depending on the amount of fragmentation before reaching the allocation point. 
Moreover, some 32-bit systems (Windows) limit user space to 2 GB. Even Linux will typically reserve the end of the address space for kernel, leaving just the rest to user space.
So we need some margin.

Reply:
Will add more to the comment, thank you for the explanation.